docs/security.html

```diff
@@ -1205,6 +1205,447 @@ <h4><a id="security_authz_cli" href="#security_authz_cli">Command Line Interface
         </tr>
     </tbody></table>
 
+    <h4><a id="security_authz_primitives" href="#security_authz_primitives">Privilege primitives</a></h4>
+    <p>Protocol calls are usually performing some operations on certain resources in Kafka. It is required to know the
+    operations and resources to set up an effective protection. In this section we'll list these operations and
+    resources, then list the combination of these with the protocols to see the valid scenarios.</p>
+    <h5>Operations in Kafka</h5>
+    <p>There are a few operation primitives that can be used to build up privileges. These can be matched up with
+    certain resources to allow specific protocol calls for a given user. These are:</p>
+    <ul>
+        <li>Read</li>
+        <li>Write</li>
+        <li>Create</li>
+        <li>Delete</li>
+        <li>Alter</li>
+        <li>Describe</li>
+        <li>ClusterAction</li>
+        <li>DescribeConfigs</li>
+        <li>AlterConfigs</li>
+        <li>IdempotentWrite</li>
+        <li>All</li>
+    </ul>
+    <h5>Resources in Kafka</h5>
+    <p>Operations above can be applied on certain resources which are described below.</p>
+    <ul>
+        <li><b>Topic:</b> this simply represents a Topic. All protocol calls that are acting on topics (such as reading,
+        writing them) require the corresponding privilege to be added. If there is an authorization error with a
+        topic resource, then a TOPIC_AUTHORIZATION_FAILED (error code: 29) will be returned.</li>
+        <li><b>Group:</b> this represents the consumer groups in the brokers. All protocol calls that are working with
+        consumer groups, like joining to a group must have privileges with the group in subject. If the privilege is not
+        given then a GROUP_AUTHORIZATION_FAILED (error code: 30) will be returned in the protocol response.</li>
+        <li><b>Cluster:</b> this resource represents the cluster. Operations that are affecting the whole cluster, like
+        controlled shutdown are protected by privileges on the Cluster resource. If there is an authorication problem
+        on a cluster resource, then a CLUSTER_AUTHORIZATION_FAILED (error code: 31) will be returned.</li>
+        <li><b>TransactionalId:</b> this resource represents the resource which represents the transactions. Actions,
```

Comment:
Would the same meaning be preserved if we say "this resource represents actions related to transactions, such as committing", instead of the first two sentences?

Reply:
yea I think so. reading my phrasing again seems a bit repetitive, I'll use yours.